lib/pbench/server/api/resources/api_key.py

```diff
@@ -63,15 +117,52 @@ def _post(
             raise APIInternalError(str(e)) from e
 
         try:
-            key = APIKey(api_key=new_key, user=user)
+            key = APIKey(api_key=new_key, user=user, name=name)
             key.add()
             status = HTTPStatus.CREATED
         except DuplicateApiKey:
             status = HTTPStatus.OK
         except Exception as e:
             raise APIInternalError(str(e)) from e
-
-        context["auditing"]["attributes"] = {"key": new_key}
-        response = jsonify({"api_key": new_key})
+        context["auditing"]["attributes"] = {"key_id": key.id, "name": name}
+        response = jsonify({"api_key": new_key, "id": key.id})
```

Comment:
I don't disagree with any of this, but....

> I think `id` is the natural react table key

Sure, but I don't expect the client to be building a table from the creation operation.  I suppose that it could try to add the newly-created key to an existing table, but it would probably be easier to do another `GET` to provide the updated table contents (and I _would_ expect the Server to return the IDs for that request).

> I suppose the UI could use the `api_key` as the table key, but `id` seems more natural.

I'm not really sure what table keys are used for, and I'm not sure that humans ever look a them.  (Mostly, they seem to be an annoyance to developers when they are missing or fail to be unique....)  That aside, I concur.

> Furthermore, we've made it the formal resource ID (e.g, to `DELETE` the key), so the client needs to know it ...

I agree, although I don't expect the client to save the value returned by the creation so that it can proceed to delete it.  I fully expect that creation and deletion will be disconnected events/processes, with the latter provided for by a `GET` request.